internal/providers/openstack/openstack.go

```diff
@@ -166,14 +172,84 @@ func fetchConfigFromDevice(logger *log.Logger, ctx context.Context, path string)
 	return os.ReadFile(filepath.Join(mnt, configDriveUserdataPath))
 }
 
+func isIPv6Address(ip net.IP) bool {
+	isIPv6 := ip.To4() == nil
+	return isIPv6
+}
+
+func findZoneID() (string, error) {
+	fmt.Println("Fetching zone id...")
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return "", fmt.Errorf("error fetching zone id: %v", err)
```

Comment:
I think it might be more appropriate to say network interfaces as this is not really the zone id, and could confuse debugging. wdyt?

"error fetching network interfaces: %v", err